src/legacy/core_plugins/data/public/query/query_bar/components/query_bar_input.tsx

```diff
@@ -29,12 +29,12 @@ import {
   getAutocompleteProvider,
 } from 'ui/autocomplete_providers';
 import { debounce, compact, isEqual, omit } from 'lodash';
-import { IndexPattern, StaticIndexPattern } from 'ui/index_patterns';
 import { PersistedLog } from 'ui/persisted_log';
 import chrome from 'ui/chrome';
 import { kfetch } from 'ui/kfetch';
 import { Storage } from 'ui/storage';
 import { localStorage } from 'ui/storage/storage_service';
+import { IndexPattern, StaticIndexPattern } from '../../../index';
```

Comment:
In looking at this again, I actually think for clarity we could instead do `'../../../../public'`

But to answer your question -- yeah, I think this could be a helpful approach specifically for _static_ stuff. I still prefer to pass down runtime dependencies via the `setup` methods of each particular service.

But the nice thing about importing static stuff this way is that it keeps things consistent with the new platform rule of "only export static code at the top level", as we are essentially importing it from the exact same place as any other plugin would.